src/server/graphql/mutations/__tests__/removeTeamMember.test.js

```diff
@@ -0,0 +1,88 @@
+import MockDate from 'mockdate';
+import mockAuthToken from 'server/__tests__/setup/mockAuthToken';
+import {__now} from 'server/__tests__/setup/mockTimes';
+import removeTeamMember from 'server/graphql/mutations/removeTeamMember';
+import MockDB from 'server/__tests__/setup/MockDB';
+import MockPubSub from 'server/__mocks__/MockPubSub';
+import fetchAndSerialize from 'server/__tests__/utils/fetchAndSerialize';
+import getRethink from 'server/database/rethinkDriver';
+import {auth0ManagementClient} from 'server/utils/auth0Helpers';
+import DynamicSerializer from 'dynamic-serializer';
+import * as getPubSub from 'server/utils/getPubSub';
+import expectAsyncToThrow from 'server/__tests__/utils/expectAsyncToThrow';
+
+MockDate.set(__now);
+console.error = jest.fn();
+
+describe('removeTeamMember', () => {
+  test('promotes another member if the person removed was the lead', async () => {
+
+  });
+
+  test('removes the teamMember, reassigns active projects to the lead', async () => {
+    // SETUP
+    const r = getRethink();
+    const mockPubSub = new MockPubSub();
+    getPubSub.default = () => mockPubSub;
+    const mockDB = new MockDB();
+    const dynamicSerializer = new DynamicSerializer();
+    await mockDB
+      .init()
+      .newProject();
+    auth0ManagementClient.__initMock(mockDB.db);
+    auth0ManagementClient.users.updateAppMetadata.mockReset();
+    const userToBoot = mockDB.db.user[7];
```

Comment:
`mockDB.db.user[7]`?

Reply:
when you call `mockDB.init()` it creates a team with 8 users on it. many tests run in parallel and unique IDs change every run so as not to conflict with records in the test database. for tests, since you know every entity that you put in a table, it's pretty easy to just use an iterator as a way to ID them. we could make a helper function like mockDB.getUserByName('steve'), but then we'd have to do that for every entity, and what are you gonna do for projects? it's not like youre gonna write a stringified JSON to identify the 1 & only project. definitely open to cleaner solutions!

Reply:
I tend to pretty much always prefer explicitly declaring state over calling procedures that statically do that work for me. For example, I'd create my two users, org, team, and project. When that gets cumbersome, I'd create a helper to do the same thing in fewer lines of code, but such that it still requires me to declare my data ahead of time.

Not a blocker, btw!

Reply:
@dan-f oh man, I'm right there with you, so let's walk down that path:
- i declare every field on every document i want in the database & then push them
- Each test is literally 200 LOCs, so i build a helper to do things like auto-populate the createdAt fields, etc.
- for my own sanity, i build another helper to ensure that teamMember.userId foreign key matches the primary key on the user document

... i just built the MockDB :wink:

Reply:
Yeah, I think I'm basically just making an argument against magic numbers and implicit state here. I understand that `MockDB.init()` is terse, but my preference is to generally be more explicit when managing global state. That's all!